### PR TITLE
Fix GCC warnings

### PR DESCRIPTION
**Pull request by Komzpa**
Patch by Darafei Praliaskouski <me@komzpa.net>

Obey const pointer in autofix logic.

### CODE REVIEW

File: liblwgeom/lwgeom_geos.c [hunk 1]
```diff
@@ -273,18 +274,24 @@ ptarray_to_GEOSLinearRing(const POINTARRAY *pa, int autofix)
...
-			pa = ptarray_addPoint(pa, getPoint_internal(pa, 0), FLAGS_NDIMS(pa->flags), pa->npoints);
-			npa = pa;
+			npa = ptarray_addPoint(pa, getPoint_internal(pa, 0), FLAGS_NDIMS(pa->flags), pa->npoints);
 		}
 		/* Check ring for having at least 4 vertices */
-		while ( pa->npoints < 4 )
+		while ( npa->npoints < 4 )
 		{
-			ptarray_append_point(pa, getPoint_internal(pa, 0), LW_TRUE);
+			ptarray_append_point(npa, (POINT4D*)getPoint_internal(npa, 0), LW_TRUE);
```
Comment (pramsey) [inline liblwgeom/lwgeom_geos.c]: Would you use getPoint4d_p instead? Getting addicted to pointing right into the array is a good way to start reading off the end of it.

**Comment by pramsey** [comment]
Very not good, in `make check`
```
  Test: test_lwgeom_clip_by_rect ...make[1]: *** [check] Segmentation fault: 11
```

**Comment by pramsey** [comment]
Why all this mucking around screwing w/ const inputs in ptarray_to_GEOSLinearRing when in ptarray_to_GEOSCoordSeq you can create and return a coordseq of the desired size without mutating the input (and, btw, breaking the const contract!)

**Comment by Komzpa** [comment]
@pramsey thanks! I did as you suggested. The only warning left is fixed in separate PR, https://github.com/postgis/postgis/pull/161

**Comment by strk** [comment]
Trac ticket: https://trac.osgeo.org/postgis/ticket/3900

**Comment by Komzpa** [comment]
Committed

**Pull request closed without merging**